backend/db/repository/posts_repo.py

```
@@ -0,0 +1,89 @@
+    
+    :param post: Description
+    :type post: Dict[str, Any]
+    :param post_content: Description
+    :type post_content: Dict[str, Any]
+    """
+    try:
+        # Try to add post in posts db table
+        res = supabase.table("posts").insert(post).execute()
+        
+        if res.error:
+            raise Exception(f"Error adding post: {res.error.message}")
```

Ruff (0.14.8)

Abstract `raise` to an inner function

(TRY301)